audioipc/src/messages.rs

```diff
@@ -175,17 +175,115 @@ fn opt_str(v: Option<Vec<u8>>) -> *mut c_char {
     }
 }
 
+#[derive(Debug)]
+pub enum RemoteHandle {
+    // Initial state on sender - live handle.
+    LocalOnSender(PlatformHandle, u32),
+    // Final state on receiver - live handle.
+    LocalOnReceiver(PlatformHandle),
+    // Transitional IPC states.
+    RemoteOnSender(PlatformHandleType),
+    RemoteOnReceiver(PlatformHandleType),
+    Empty,
+}
```

Comment:
This looks like a great idea!

Would you mind adding more comments about how the handle is transferred, in Windows and Unix respectively? I'd hope some handle ownership (e.g., who will close the handle) can be documented here.